bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml

```diff
@@ -64,13 +64,25 @@ spec:
           image: {{ include "influxdb.volumePermissions.image" . }}
           imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
           command:
-            - sh
-            - -c
+            - /bin/bash
+            - -cx
             - |
+              {{- if .Values.persistence.enabled }}
               mkdir -p /bitnami/influxdb/{data,meta,wal}
-              chown -R "{{ .Values.influxdb.securityContext.runAsUser }}:{{ .Values.influxdb.securityContext.fsGroup }}" /bitnami/influxdb/{data,meta,wal}
+              chmod 700 /bitnami/influxdb/{data,meta,wal}
+              find /bitnami/influxdb/{data,meta,wal} -mindepth 1 -maxdepth 1 | \
+              {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
+                xargs chown -R `id -u`:`id -G | cut -d " " -f2`
+              {{- else }}
+                xargs chown -R {{ .Values.influxdb.securityContext.runAsUser }}:{{ .Values.influxdb.securityContext.fsGroup }}
+              {{- end }}
```

Comment:
Why using this `find .... | xargs chown -R ...` approach? This approach can be useful when you're excluding some directories or files, but I think that's not the case.

I mean, is there any actual difference between this and `chown -R ... /bitnami/influxdb/{data,meta,wal}`

Reply:
Removed the `xargs`, not necessary